app/RouteHandlers/PageDisplayHandler.php

```diff
@@ -63,8 +63,16 @@ public function getPageTemplate( string $pageId, array $context = [] ): TwigTemp
 
 		if ( file_exists( $this->ffFactory->getAbsoluteSkinDirectory() . DIRECTORY_SEPARATOR . $pageTemplate ) ) {
 			$template = $pageTemplate;
+			$context = $this->getAdditionalContextForPageId( $pageId, $context );
```

Comment:
👍 
I like that you did not modify `$context` in-place but return it! That makes what's happening much more explicit and readable.